cmd/queue/main.go

```diff
@@ -155,9 +157,17 @@ func proxyHandler(reqChan chan queue.ReqEvent, breaker *queue.Breaker, tracingEn
 
 		// Enforce queuing and concurrency limits.
 		if breaker != nil {
+			cf := noop
+			if tracingEnabled {
+				ctx, waitSpan := trace.StartSpan(r.Context(), "queueWait")
+				r = r.WithContext(ctx)
+				cf = waitSpan.End
+			}
 			if err := breaker.Maybe(r.Context(), func() {
+				cf()
```

Comment:
I would create a dummy `func() error` containing the span creation, a defer span.End and a call to breaker.Maybe.
Even better, there should be a `Wait method` abstracting this entirely.
Just
err := wait()
if err != nil {
...
}
next.ServeHTTP

Reply:
Same as above. If you have a good idea how to wrap `Maybe` — send a PR, I'd gladly approve :)

Reply:
Github also supports multi-line suggested edits now 😇

Reply:
@mattmoor too much work 😢 
Ok, merge it. I'll take a look later if it can be made cleaner.